### PR TITLE
Implement various fixes and additional features

### PR DESCRIPTION
**Pull request by jochenvdv**
- [x] Pin structlog dependency (because using private function that is not part of public structlog API and can change without notice)
- [x] Set minimum logging level to DEBUG when in dev environment
- [x] Log uncaught exceptions
- [x] Intercept & process STDLIB logging
- [x] Populate type field dynamically
- [x] Populate source field dynamically
- [x] Refactor into functions
- [x] Allow binding request ID field
- [x] Allow binding component field
- [x] Disable JSON logging & enable pretty logging when executing locally

### CODE REVIEW

File: novemberfive_logging/_logger.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    elif os.getenv("ENV") == 'dev':
+        log_level = logging.DEBUG
+    else:
+        log_level = logging.INFO
+
+    return log_level
+
+
+def _get_renderer():
+    if "ENV" not in os.environ:
```
Comment (jorenvh1) [inline novemberfive_logging/_logger.py]: ik zou hier nog "or os.getenv("ENV") == 'dev'" aan toevoegen?

File: tests/test_logger.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    configure_logging(component="test")
+    log = structlog.get_logger()
+
+    try:
+        raise RuntimeError("Oops")
+    except RuntimeError as e:
+        log.critical("Something bad happened!", exception=e)
+
+    captured_output = capsys.readouterr()
+    print(captured_output.out)
```
Comment (jorenvh1) [inline tests/test_logger.py]: die prints mogen hier weg denk ik?

**Review by jorenvh1 — CHANGES_REQUESTED** [review]

File: novemberfive_logging/_logger.py (hunk 1 above)
Comment (jochenvdv) [reply novemberfive_logging/_logger.py]: Dat kan, heb ik over nagedacht. Willen we pretty logging als env DEV is? Die logging is specifiek geformat voor de terminal, ik denk dat we in de echte develop omgeving best JSON formaat aanhouden.

File: tests/test_logger.py (hunk 2 above)
Comment (jochenvdv) [reply tests/test_logger.py]: Klopt, achtergebleven na debuggen 😆

File: novemberfive_logging/_logger.py (hunk 1 above)
Comment (jochenvdv) [reply novemberfive_logging/_logger.py]: Als ENV='dev' betekent die lokale omgeving, dus deze change neem ik nog mee.

**Pull request merged**